tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -771,43 +771,43 @@ void TangoRosNode::StopPublishing() {
   if (run_threads_) {
     run_threads_ = false;
     if (device_pose_thread_.publish_thread.joinable()) {
-      if (!tango_data_available_ || !publish_pose_on_tf_
-          || !publish_pose_on_topic_) {
+      if (!tango_data_available_ || !publish_pose_on_tf_ ||
+          !publish_pose_on_topic_) {
```

Comment:
In this case it is because we call [wait](http://en.cppreference.com/w/cpp/thread/condition_variable/wait), which block the publisher thread until `notify_all()` is called.
Normally `notify_all()` is called in the Tango callback but in some conditions the call to `notify_all()` is not reached.